doc/dev/hooks.md

```diff
@@ -62,6 +62,37 @@ The following functions **may** be exported (optional):
   - Signature: `(Plugin * handle, KeySet * returned, Key * parentKey)`
   - Called in `kdbSet` after the storage phase.
 
+### `record` hook
+
+Used for the session recording plugin.
+Hard coded to search for a plugin named `recorder`.
+
+The following function must be exported:
+
+- `record`:
+
+  - Signature: `(Plugin * handle, KeySet * returned, Key * parentKey)`
+  - Called in `kdbSet` after the storage phase.
+  - Must not modify the `returned` keyset.
+  - The `parentKey` must not be modified, except for adding errors and warnings.
+  - Calculates the changes and stores them.
+
+- `lock`:
```

Comment:
This should be in separate locking plugins.

Reply:
Well it has quite some specific code for recording in it, to detect whether we really need to lock or not. I'm not sure that this would be quite as possible if we just use generic locking plugins.